prow/spyglass/lenses/common/common.go

```diff
@@ -228,24 +235,30 @@ type ProwJobFetcher interface {
 
 // prowToGCS returns the GCS key corresponding to the given prow key
 // TODO: Unexport once we only have remote lenses
-func ProwToGCS(fetcher ProwJobFetcher, config config.Getter, prowKey string) (string, error) {
+func ProwToGCS(fetcher ProwJobFetcher, config config.Getter, prowKey string) (string, string, error) {
 	jobName, buildID, err := keyToJob(prowKey)
 	if err != nil {
-		return "", fmt.Errorf("could not get GCS src: %v", err)
+		return "", "", fmt.Errorf("could not get GCS src: %v", err)
 	}
 
 	job, err := fetcher.GetProwJob(jobName, buildID)
 	if err != nil {
-		return "", fmt.Errorf("Failed to get prow job from src %q: %v", prowKey, err)
+		return "", "", fmt.Errorf("failed to get prow job from src %q: %v", prowKey, err)
 	}
 
 	url := job.Status.URL
 	prefix := config().Plank.GetJobURLPrefix(job.Spec.Refs)
 	if !strings.HasPrefix(url, prefix) {
-		return "", fmt.Errorf("unexpected job URL %q when finding GCS path: expected something starting with %q", url, prefix)
+		return "", "", fmt.Errorf("unexpected job URL %q when finding GCS path: expected something starting with %q", url, prefix)
 	}
-	return url[len(prefix):], nil
-
+	// we parse the storageProvider from the last jobURLPrefix segment, e.g.:
+	// https://prow.k8s.io/view/gcs/ => gcs
+	urlPrefixSegments := strings.Split(strings.TrimSuffix(prefix, "/"), "/")
```

Comment:
@alvaroaleman I think we need something better then this as it only works if the jobURLPrefix ends with gcs/gs/s3. Do you have an idea how we can/should detect the storage provider? `ProwToGCS` is used when we handle `prowjob` urls

Reply:
It it possible to do something like this:
* `prowjobs/<storage-provider>/<path>`
* `prowjobs/<path>`  (fall back to gs)

Do you know where these prowjobs urls come from?

Reply:
What do you think about using the job retrieved in l.244 and using the storageProvider discovered from prowjob.Spec.DecorationConfig.GCSConfiguration.Bucket 
(if prefix found => use that one, if not gs)

Reply:
yeah, that sgtm.

Reply:
Done